lib/liquid/block.rb

```diff
@@ -13,10 +20,22 @@ def parse(tokens)
       @blank = true
       @nodelist ||= []
       @nodelist.clear
+      @@whitespace ||= false
 
       while token = tokens.shift
         case token
         when IsTag
+          @@whitespace = false
+          if token =~ IsWhiteSpaceTagEnd
+            @@whitespace = true
+          end
+          if token =~ IsWhiteSpaceTagStart
```

Comment:
Using a regex to check a single byte seems like overkill if you are worried about performance.  E.g. `s.getbyte(2) == TRIM_CHAR` where `TRIM_CHAR = '-'.ord`